research.rst

```diff
@@ -0,0 +1,315 @@
+Tequila
+=======
+
+What is Tequila?
+
+- it's a collection of Ansible roles
+- it's a wrapper script providing a simplified interface for the
+  ``ansible`` command-line program
+
+
+New Projects
+------------
+
+Creation of new projects would be done using django-project-template,
+just as we currently do for margarita.  Installing tequila would make
+available the ``tequila`` command, providing a simplified interface
+for ``ansible-playbook`` (and friends) that will be mostly familiar to
+current users of the django-project-template fabric scripts.  There
+would also be some mild changes to the project structure, i.e. the
+location of the project-specific variables and custom playbooks, since
+these would be kept distinct from paths where they might conflict with
+the equivalent Salt files.
+
+Much of this work has already been started, and is currently on the
+``use_tequila`` branch.
+
+
+Installation
+------------
+
+We need to have the ability to pin projects to particular versions of
+tequila.  It is also desirable to have an easy command to install or
+upgrade tequila, along the lines of ``pip install``.
+
+Under all of the proposals that follow, I recommend that Tequila be
+pip installable.  The work for this has already been done.
+
+
+Proposal 1: ``tequila roles`` command puts the roles in the right place
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Under this proposal, no changes to the structure of the Tequila
+repository need to happen.  Running the ``tequila roles`` command
+would symlink or copy the ``tequila/roles`` directory up to the
+current working directory (which under most scenarios would be the
+project directory), and this would generally not need to be run more
+than once per local setup.
+
+For development of the roles themselves, one could ``pip install -e
+<local_tequila_copy>``, and the in-progress versions of the roles
+would easily be made available for the ansible commands.
+
+The pros:
+
+- no changes to our Tequila repo structure
+- could still run plain versions of ``ansible-playbook``
+
+The cons:
+
+- our roles would not be easily usable by members of the Ansible
+  community
```

Comment:
How about if django-project-template created a `roles` directory, and `tequila roles` just symlinked individual tequila roles into that directory (not overwriting any roles that might already exist)?  That way all the roles would be under `./roles`, and a project could override any role by putting its own copy under `roles`.